src/Tools/dotnet-monitor/Exceptions/ExceptionsOperation.cs

```diff
@@ -85,12 +87,51 @@ public Task StopAsync(CancellationToken token)
 
         private async Task WriteJson(Stream stream, IReadOnlyList<IExceptionInstance> instances, CancellationToken token)
         {
-            foreach (IExceptionInstance instance in instances)
+            foreach (IExceptionInstance instance in FilterExceptions(_configuration, instances))
```

Comment:
Same issue in https://github.com/dotnet/dotnet-monitor/pull/5131/files#r1300421481, but this is more problematic because we'd be reporting an ID for an exception that doesn't exist in the output.

Reply:
This is interesting...I actually viewed that as a feature and not a bug, but I guess it depends on how people would expect to use this. My thinking was that if we included the associated inner exceptions, that means users are seeing things that didn't match their filter's spec (and if they wanted to see a broader swath that might include the inner exceptions, you'd just expand your filter). I guess you could either be annoyed that there's a lot of inner exceptions that you didn't ask for in your filter, or you could be annoyed that you only have the top-level exceptions and you need to adjust your query (or GET /exceptions) to see the inner ones...

This starts to feel like you _could_ offer a knob for whether you want to include inner exceptions, but I'm not sure that we want to do that (at least not initially). If you weren't sure, it's probably safer from a breaking perspective to only include the top-level and then add in inner exceptions later if people ask - the other direction involves yanking out things that previously showed up (or adding a knob). But, if we know we definitely want the inner exceptions, then we can include them now.